src/extension/app/components/plugin/env-switcher/env-switcher.js

```diff
@@ -279,15 +279,13 @@ export class EnvironmentSwitcher extends MobxLitElement {
 
   /**
    * Handles the environment switcher change event
-   * @param {Event} event - The change event
    */
-  onChange(event) {
+  onChange() {
     const { picker } = this;
     const { value } = picker;
 
-    // TODO: Figure out how to get keyboard state
-    // @ts-ignore
-    appStore.switchEnv(value, newTab(event));
+    const openNewTab = value === 'edit' ? true : newTab(appStore.keyboardListener);
```

Comment:
Since this is triggered by a `change` event in the picker and not a keyboard or mouse event we don't have access to the current state of the keyboard. Setup a keyboard listener which can help understand the state of the keyboard in these cases.

Reply:
Isn't the change event a `MouseEvent` under the hood? If so, you'd have a `key` property...

Reply:
Unforetunly it's not. This is all we get.
![Screenshot 2024-03-08 at 12 58 13 PM](https://github.com/adobe/aem-sidekick/assets/3231084/936239ec-2f15-4623-879d-bdf4295ac314)